src/Orchard.Web/Modules/Orchard.Projections/Handlers/FieldIndexPartHandler.cs

```diff
@@ -43,24 +46,56 @@ public class FieldIndexPartHandler : ContentHandler {
                 context.Builder.Weld<FieldIndexPart>();
             }
         }
+        private void Updated(UpdateContentContext context, FieldIndexPart fieldIndexPart) {
+            if (context.UpdatingItemVersionRecord.Latest) { // updates projection draft indexes only if it is the latest version
+                foreach (var part in fieldIndexPart.ContentItem.Parts) {
```

Comment:
Try to externalize this method, and use a lambda as the differences.

Reply:
OK, I will try...